java-manta-client/src/main/java/com/joyent/manta/client/MantaSeekableByteChannel.java

```diff
@@ -178,11 +174,14 @@ public int read() throws IOException {
             throw new ClosedChannelException();
         }
 
-        final HttpResponse response = connectOrGetResponse();
-        final HttpEntity entity = response.getEntity();
-        final InputStream is = entity.getContent();
-        position++;
-        return is.read();
+        final MantaObjectInputStream stream = connectOrGetResponse();
+        final int read = stream.read();
+
+        if (read > -1) {
+            position++;
```

Comment:
Great catch. I got atomicity and memory visibly confused. For some reason, I misremembered volatile as AtomicX for primitives. This is certainly not true. I've updated to use a AtomicLong.